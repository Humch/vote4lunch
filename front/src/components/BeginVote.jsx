import React from 'react';

import { Container, Row, Col, Button } from 'reactstrap'
import { Link } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faPlus } from '@fortawesome/free-solid-svg-icons'

const BeginVote = () => {
    return ( 
        <Container  fluid className="stepContainer">
            <Row noGutters className="justify-content-center align-items-center h-100">
                <Col
                    xs="4" 
                    className="bg-blue p-5 text-center rounded"
                >
                <p className="text-white">Faites voter vos collègues pour le déjeuner</p>
                    <Button tag={Link} to="/begin-vote" color="success">Organiser un vote <FontAwesomeIcon icon={faPlus} /></Button>
                </Col>
            </Row>
        </Container>
     );
}
 
export default BeginVote;