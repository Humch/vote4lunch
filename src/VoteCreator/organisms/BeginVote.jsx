import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { Container, Row, Col, Button } from 'reactstrap';
import { Link } from 'react-router-dom';

import { updateUserData } from '../../Accounts/actions';

class BeginVote extends Component {
  componentDidMount() {
    const pseudo = localStorage.getItem('pseudo');
    const email = localStorage.getItem('email');

    if (pseudo && email) {
      this.props.updateUserData(pseudo, email);
    }
  }

  render() {
    return (
      <Container fluid className="BeginVote">
        <Row noGutters className="justify-content-center align-items-center h-100">
          <Col
            xs="12"
            md="4"
            className="bg-blue p-5 text-center rounded"
          >
            <p className="text-white">Fait voter tes collègues pour le déjeuner</p>
            <Button tag={Link} to="/create-a-vote" color="success">Organiser un vote</Button>
          </Col>
        </Row>
      </Container>
    );
  }
}

BeginVote.propTypes = {
  updateUserData: PropTypes.func.isRequired,
};

const mdtp = dispatch => bindActionCreators({ updateUserData }, dispatch);

export default connect(null, mdtp)(BeginVote);
