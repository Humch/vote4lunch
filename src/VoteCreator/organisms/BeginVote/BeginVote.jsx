import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import {
  Container,
  Row,
  Col,
  Button,
} from 'reactstrap';
import { Link } from 'react-router-dom';

import { updateUserData } from '../../../Accounts/actions';

class BeginVote extends Component {
  componentDidMount() {
    const pseudo = localStorage.getItem('pseudo');
    const email = localStorage.getItem('email');
    const authenticatedStorage = localStorage.getItem('authenticated');

    if (pseudo && email && authenticatedStorage) {
      const { updateUserData: updateUD } = this.props;
      updateUD(pseudo, email, JSON.parse(authenticatedStorage));
    }
  }

  render() {
    const { authenticated } = this.props;

    return (
      <Container fluid className="BeginVote">
        <Row noGutters className="justify-content-center align-items-center h-100">
          <Col
            xs="12"
            md="4"
            className="bg-blue p-5 text-center rounded"
          >
            <p className="text-white">{ !authenticated ? 'Connecte toi pour créer un vote' : 'Fait voter tes collègues pour le déjeuner' }</p>
            <Button
              tag={Link}
              to="/create-a-vote"
              color="success"
              disabled={!authenticated}
            >
              {'Organiser un vote'}
            </Button>
          </Col>
        </Row>
      </Container>
    );
  }
}

BeginVote.propTypes = {
  updateUserData: PropTypes.func.isRequired,
  authenticated: PropTypes.bool.isRequired,
};

const mstp = ({ userData }) => ({
  authenticated: userData.authenticated,
});

const mdtp = dispatch => bindActionCreators({ updateUserData }, dispatch);

export default connect(mstp, mdtp)(BeginVote);
