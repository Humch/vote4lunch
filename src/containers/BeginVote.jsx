import React, {Component} from 'react';
import {connect} from 'react-redux'

import { Container, Row, Col, Button } from 'reactstrap'
import { Link } from 'react-router-dom';

import {updateUserData} from '../actions'

class BeginVote extends Component {

    componentDidMount() {
        const pseudo = localStorage.getItem('pseudo')
        const email = localStorage.getItem('email')

        if (pseudo && email) {
            this.props.dispatch(updateUserData(pseudo, email))
        }
    }

    render() { 
        return ( 
            <Container fluid className="BeginVote">
            <Row noGutters className="justify-content-center align-items-center h-100">
                <Col
                    xs="12"
                    md="4"
                    className="bg-blue p-5 text-center rounded"
                >
                <p className="text-white">Fait voter tes collègues pour le déjeuner</p>
                    <Button tag={Link} to="/create-a-vote" color="success">Organiser un vote</Button>
                </Col>
            </Row>
        </Container>
         );
    }
}
 
export default connect()(BeginVote);