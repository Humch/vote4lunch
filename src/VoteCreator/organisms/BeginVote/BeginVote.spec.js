/* eslint-disable no-undef */
import React from 'react';
import { Provider } from 'react-redux';

import { BrowserRouter as Router } from 'react-router-dom';

import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

// mock browser local storage
import LS from '../../../../__mocks__/localStorage';

import BeginVote from './BeginVote';

describe('BeginVote Snapshot', () => {
  const initialState = {
    userData: {
      authenticated: false,
    },
  };
  const mockStore = configureStore();
  let store;

  beforeEach(() => {
    store = mockStore(initialState);
  });

  afterEach(() => {
    LS.window.localStorage.clear();
  });

  it('capturing Snapshot of BeginVote, user is not logged in', () => {
    const renderedValue = renderer.create(
      <Router>
        <Provider store={store}>
          <BeginVote />
        </Provider>
      </Router>
    ).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });

  it('capturing Snapshot of BeginVote, user is logged in', () => {
    LS.window.localStorage.setItem('pseudo', 'bob');
    LS.window.localStorage.setItem('email', 'bob@bob.com');
    LS.window.localStorage.setItem('authenticated', 'true');

    const renderedValue = renderer.create(
      <Router>
        <Provider store={store}>
          <BeginVote />
        </Provider>
      </Router>
    ).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});

describe('Beginvote (Shallow + passing the {store} directly)', () => {
  const initialState = {
    userData: {
      authenticated: false,
    },
  };
  const mockStore = configureStore();
  let store;
  let container;

  beforeEach(() => {
    store = mockStore(initialState);
    container = shallow(<BeginVote store={store} />);
  });

  afterEach(() => {
    LS.window.localStorage.clear();
  });

  it('render the connected(SMART) component', () => {
    expect(container.length).toEqual(1);
  });

  it('check Prop matches with initialState', () => {
    expect(container.prop('authenticated')).toEqual(initialState.userData.authenticated);
  });
});

describe('Beginvote REACT-REDUX (Mount + wrapping in <Provider>) - user is logged in', () => {
  const initialState = {
    userData: {
      authenticated: true,
    },
  };
  const mockStore = configureStore();
  let store;
  let wrapper;

  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = mount(
      <Router>
        <Provider store={store}>
          <BeginVote />
        </Provider>
      </Router>
    );
  });

  afterEach(() => {
    LS.window.localStorage.clear();
  });

  it('render the connected(SMART) component', () => {
    expect(wrapper.length).toEqual(1);
  });
});
