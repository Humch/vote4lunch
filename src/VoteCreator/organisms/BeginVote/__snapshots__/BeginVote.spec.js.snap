// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeginVote Snapshot capturing Snapshot of BeginVote, user is logged in 1`] = `
<div
  className="BeginVote container-fluid"
>
  <div
    className="justify-content-center align-items-center h-100 no-gutters row"
  >
    <div
      className="bg-blue p-5 text-center rounded col-12 col-md-4"
    >
      <p
        className="text-white"
      >
        Connecte toi pour créer un vote
      </p>
      <a
        aria-label={null}
        className="btn btn-success disabled"
        disabled={true}
        href="/create-a-vote"
        onClick={[Function]}
      >
        Organiser un vote
      </a>
    </div>
  </div>
</div>
`;

exports[`BeginVote Snapshot capturing Snapshot of BeginVote, user is not logged in 1`] = `
<div
  className="BeginVote container-fluid"
>
  <div
    className="justify-content-center align-items-center h-100 no-gutters row"
  >
    <div
      className="bg-blue p-5 text-center rounded col-12 col-md-4"
    >
      <p
        className="text-white"
      >
        Connecte toi pour créer un vote
      </p>
      <a
        aria-label={null}
        className="btn btn-success disabled"
        disabled={true}
        href="/create-a-vote"
        onClick={[Function]}
      >
        Organiser un vote
      </a>
    </div>
  </div>
</div>
`;
